src/codeu/chat/client/ClientUser.java

```diff
@@ -44,19 +45,21 @@ public ClientUser(Controller controller, View view) {
     this.view = view;
   }
 
-  // Validate the username string
-  static public boolean isValidName(String userName) {
-    boolean clean = true;
-    if (userName.length() == 0) {
-      clean = false;
-    } else {
 
-      // TODO: check for invalid characters
+  // Makes sure the username/password combo doesn't contain any bad chars
+  static public boolean isValidUserInput(String username, String password) {
 
-    }
-    return clean;
+    return isValidInput(username) && isValidInput(password);
   }
 
+
+  // Validate a user input string
+  static public boolean isValidInput(String userInput) {
+
```

Comment:
nit: extra white space

I'd also add a comment explaining what this regular expression means.